### PR TITLE
Deprecate DatabaseTestCase.php

### PR DESCRIPTION
**Pull request by gesinn-it-ilm**
Refactored tests for SMW. 
Added _SMWIntegrationTestCase.php_ - This class wraps **MediaWikiIntegrationTestCase** for usage with SMW tests.
**(this class should replace DatabaseTestCase)**

Updated phpunit.xml.dist where new testsuites were added, tests devided and organised better for running in CI. 
Added new testsuits to main.yml (CI updated).

CI is running now without errors in the logs. Still has deprecation warnings and failing tests which need to be checked and fixed. 
(CI run manually on separate branch for testing)

## Summary by CodeRabbit

- **Refactor**
  - Optimized PHPUnit test suite configuration in the GitHub Actions workflow for better test execution order.
  - Updated test suite declarations in `phpunit.xml.dist` for improved test organization.

- **Tests**
  - Refactored various test cases to inherit from improved base test classes for better testing practices and code maintainability.

### CODE REVIEW

**Comment by gesinn-it-gea** [comment]
@JeroenDeDauw , @kghbln fyi: this is not yet finished but should be part of the next SMW release (4.2.0?)

**Comment by JeroenDeDauw** [comment]
Would be nice to get this into the release, yes.

**Comment by YOUR1** [comment]
When do you expect this to be ready? Since the release of 4.2.0 is now depending on it?

**Comment by gesinn-it-gea** [comment]
@JeroenDeDauw ... is already in progress. The current CI setup gives a false picture of the code base, as not all tests, but only selected "test suites" are run: https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/.github/workflows/main.yml#L123 and https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/.github/workflows/main.yml#L129

I suggest to first remove those filters and fix the errors and failures in a joint approach before changing to an alternate CI and test setup. Otherwise, we will drag the legacy into the new architecture.

fyi: @gesinn-it-evl , @gesinn-it-ilm , @kghbln , @krabina

**Comment by JeroenDeDauw** [comment]
I would take the opposite approach: continue excluding the tests that we know are broken. Fixing those is something that could be worked on, but it is not necessary to get the working tests to run on more recent MediaWiki versions, which is what would provide the value for the release.

**Comment by gesinn-it-gea** [comment]
@JeroenDeDauw okay, let's do it like you suggested. I'll start a separate PR today.

File: tests/phpunit/Integration/MediaWiki/Hooks/ParserFirstCallInitIntegrationTest.php [hunk 1]
```diff
@@ -98,7 +98,8 @@ public function testParseWithParserFunctionEnabled( $parserName, $text ) {
 
 		$expectedNullOutputFor = [
 			'concept',
-			'declare'
+			'declare',
+			'ask'
```
Comment (gesinn-it-gea) [inline tests/phpunit/Integration/MediaWiki/Hooks/ParserFirstCallInitIntegrationTest.php:102]: I dont't think adding ask to "expectedNullOutputFor" is the right way to solve this. I'd expect that "ask" creates some output. This requires further investigation. We could temp. skip this test with a clear message describing the issue.

File: tests/phpunit/Integration/MediaWiki/Jobs/UpdateJobRoundtripTest.php
Comment (gesinn-it-gea) [inline tests/phpunit/Integration/MediaWiki/Jobs/UpdateJobRoundtripTest.php:1]: I think this test needs to be discussed. I'm not sure whether the test actually checks what the original idea was.

File: tests/phpunit/Integration/MediaWiki/MediaWikiIntegrationForRegisteredHookTest.php [hunk 2]
```diff
@@ -61,6 +61,7 @@ protected function tearDown() : void {
 		$this->applicationFactory->clear();
 		$this->mwHooksHandler->restoreListedHooks();
 
+		$this->title = Title::newFromText( __METHOD__ );
```
Comment (gesinn-it-gea) [inline tests/phpunit/Integration/MediaWiki/MediaWikiIntegrationForRegisteredHookTest.php:64]: why do we first create a page and then delete it in tearDown()?. Do we need the two lines at all? I'd expect that clean-up is done by MW's test framework now.

File: tests/phpunit/Integration/SQLStore/SubSemanticDataDBIntegrationTest.php [hunk 3]
```diff
@@ -41,6 +41,7 @@ protected function setUp(): void
 	protected function tearDown() : void {
 		$pageDeleter= new PageDeleter();
 
+		// $this->title = Title::newFromText( __METHOD__ );
```
Comment (gesinn-it-gea) [inline tests/phpunit/Integration/SQLStore/SubSemanticDataDBIntegrationTest.php:44]: why this comment?

File: tests/phpunit/Integration/SQLStore/SubSemanticDataDBIntegrationTest.php [hunk 4]
```diff
@@ -41,6 +41,7 @@ protected function setUp(): void
...
 		$pageDeleter->deletePage( $this->title );
```
Comment (gesinn-it-gea) [inline tests/phpunit/Integration/SQLStore/SubSemanticDataDBIntegrationTest.php:45]: do we really still have to tidy up our things ourselves?

**Comment by gesinn-it-ilm** [comment]
@gesinn-it-gea this PR is not up to date and in use any more. All of our work regarding this topic is moved to another branch [DatabaseTestCase_deprecated](https://github.com/SemanticMediaWiki/SemanticMediaWiki/tree/DatabaseTestCase_deprecated).
There we are working on **MediaWikiIntegrationTestCase** implementation and we have made some progress regarding integration tests.

**Comment by gesinn-it-gea** [comment]
@Seb35 ... just to make you aware of this branch. If you want to get involved, you are most welcome.

**Comment by Seb35** [comment]
> @Seb35 ... just to make you aware of this branch. If you want to get involved, you are most welcome.

Yes, I’m aware of it. I try to look in more details in the next days

**Comment by gesinn-it-ilm** [comment]
@Seb35 - this branch is not up to date and in use for quite some time. We switched all the work to another branch where you can work if you like. The branch is [DatabaseTestCase_deprecated](https://github.com/SemanticMediaWiki/SemanticMediaWiki/tree/DatabaseTestCase_deprecated).
There we are working on MediaWikiIntegrationTestCase implementation and we have made huge progress.
This PR will be closed now.

**Pull request closed without merging**